index.js

```diff
@@ -78,10 +78,16 @@ var isSupportedHash = exports.isSupportedHash = function (msg) {
 }
 
 var isSigMatchesCurve = exports.isSigMatchesCurve = function (msg) {
-  if(!isSignatureRx.test(msg.signature)) return new Error('signature was not canonical base64')
+  if(!isSignatureRx.test(msg.signature)) return
```

Comment:
Why leave out the error msg?

Reply:
Other functions expect `isSigMatchesCurve` to return truthy if the signature is valid, or falsey if the signature is invalid. Returning an error here is wrong because it's interpreted as truthy, which means "this is valid" even though the value we're returning is an error. :facepalm: 

Combined with the previous error that I've described, this function has historically always returned truthy because:

1. The regex is broken and `isSignatureRx.test(msg.signature)` is always false.
2. The error handling is broken and when `!isSignatureRx.test(msg.signature)` happens we return a truthy value.

The two bugs cancel each other out so that we don't notice them, but `isSigMatchesCurve()` has always returned truthy, even when it shouldn't.

Reply:
Also briefly documented in my commit message: d3e2dac

Reply:
omg